pkg/apiserver/registry/core/shoot/strategy_test.go

```diff
@@ -65,7 +67,77 @@ var _ = Describe("Strategy", func() {
 		})
 	})
 
+	Describe("#PrepareForCreate", func() {
+		Context("cloudProfile field removal", func() {
+			var (
+				shoot                 *core.Shoot
+				cloudProfileReference *core.CloudProfileReference
+			)
+
+			BeforeEach(func() {
+				shoot = &core.Shoot{}
+				cloudProfileReference = &core.CloudProfileReference{
+					Kind: "foo",
+					Name: "bar",
+				}
+			})
+
+			It("should remove cloudProfile field if namespacedcloudprofile feature gate is disabled", func() {
```

Comment:
You could use camel case for the feature gate name

Reply:
Done https://github.com/gardener/gardener/pull/9440/commits/8daad6c29ddd3ced99d036dbd0bc5d77cf3c9369